### ISSUE TITLE
Automatic creation of a .watched file when playing a video - with working solution suggested

### ISSUE THREAD

**Issue by Omertron** [YAMJ/yamj-v2#2424]
Original [issue 2425](https://code.google.com/p/moviejukebox/issues/detail?id=2425) created by Omertron on 2012-05-23T11:01:24.000Z:

<b>What device are you using YAMJ on? Networked Media Tank, PlayonHD, Other?</b>
NMT A-210

<b>Please describe in as much detail as possible what new feature you want:</b>
With the default skin, by pressing the red button, a .watched file is dynamically created or deleted by a JavaScript script calling a CGI script.

It would be nice that a .watched file will be created automatically when playing a video, if it does not already exist.

I already tested a working solution :
By adding an onclick event on the play link that calling the watched Javascript function with 0 value : onclick=&quot;watched(0)&quot;.

This works well when first playing a video, the .watched file is created immediately after playing the video.
The problem is at the next read : watched file will be deleted, because of Javascript watched function works like an on / off switch.

So I modified the watched.js script to accept a new value for the parameter of the watched function.
Currently the value 0 is made to create or delete the watched file, and the value 1 only for checking if the watched file exists.
With the new value 2, the watched file will be created only if it does not already exist which solves the problem.

In summary, watched javascript function should be improved (see attached watched.js example file), and onclick event calling function watched (2) should be added on all the play links (by editing detail.xsl file).

bests regards,
Guillaume
PS : Sorry for my poor English, I'm French
